src/types.ts

```diff
@@ -223,6 +225,10 @@ export type AppState = {
   pendingImageElementId: ExcalidrawImageElement["id"] | null;
   showHyperlinkPopup: false | "info" | "editor";
   selectedLinearElement: LinearElementEditor | null;
+  /** the user's clientId who is being followed on the canvas */
+  userToFollow: UserToFollow | null;
+  /** whether follow mode should be disconnected when the non-remote user interacts with the canvas */
+  shouldDisconnectFollowModeOnCanvasInteraction: boolean;
```

Comment:
Yeah, so the idea for this was that it would be up to the host app whether to enable this behaviour or not.

I discussed this with @maielo and we agreed that by default any canvas interaction should disable follow mode and this behaviour should be consistent across OSS and E+ for example.

But I personally still think making this configurable makes sense in some specific circumstances. One such idea was a *classroom environment* where students may follow a teacher around the canvas. In that case we could enable view mode for the followers as well as set this flag to false so that:
- students can't interfere with the teachers actions
- follow mode can still be disconnected explicitly by clicking on the teacher's avatar again or using the X on the badge at the bottom of the editor, but at least accidental clicks/scrolls won't disconnect the student implicitly which I think might be the desired behaviour in this instance

But if we think this is not really necessary and we don't really want to make this configurable, I'm not against removing it.